libr/core/core.c

```diff
@@ -2450,7 +2450,6 @@ R_API RCore *r_core_fini(RCore *c) {
 	r_list_free (c->files);
 	r_list_free (c->watchers);
 	r_list_free (c->scriptstack);
-	r_list_free (c->tasks);
```

Comment:
The free function saved in core->tasks is r_core_task_decref, freeing the list should just decrease the refcount for each task as you would expect. If this leads to a uaf in some case, the problem is somewhere else, maybe somewhere the refcount is not increased where it should. If you have a case where this happens, please elaborate.

Also, you still have to free the RList itself or it will leak.

Reply:
you re right this was the wrong approach but there is genuinely an UAF case. With the other diff you can see why there is.

Reply:
Yes, you are absolutely right. Thanks for noticing, the fix is now correct.